File: app/src/main/java/com/dancc/basketballstatskeeper/adapter/StatsAdapter.java
package com.dancc.basketballstatskeeper.adapter;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.dancc.basketballstatskeeper.R;
import com.dancc.basketballstatskeeper.model.GameStats;
import java.util.List;

public class StatsAdapter extends RecyclerView.Adapter<StatsAdapter.StatsViewHolder> {

  private List<GameStats> gameStats;

  public StatsAdapter() {
  }

  public void setGameStats(List<GameStats> gameStats) {
    this.gameStats = gameStats;
    notifyDataSetChanged();
  }

  @NonNull
  @Override
  public StatsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    View v = LayoutInflater.from(parent.getContext())
        .inflate(R.layout.stat_row_header_view, parent, false);

    return new StatsViewHolder(v);
  }

  @Override
  public void onBindViewHolder(@NonNull StatsViewHolder holder, int position) {
    holder.bind(gameStats.get(position));
  }

  @Override
  public int getItemCount() {
    return gameStats == null ? 0 : gameStats.size();
  }

  static class StatsViewHolder extends RecyclerView.ViewHolder {
    @BindView(R.id.pts)
    TextView pts;

    @BindView(R.id.reb)
    TextView reb;

    @BindView(R.id.ast)
    TextView ast;

    @BindView(R.id.blk)
    TextView blk;

    @BindView(R.id.stl)
    TextView stl;

    @BindView(R.id.to)
    TextView to;

    @BindView(R.id.twoPT)
    TextView twoPT;

    @BindView(R.id.threePT)
    TextView threePT;

    @BindView(R.id.onePT)
    TextView onePT;

    StatsViewHolder(View v) {
      super(v);

      ButterKnife.bind(this, v);
    }

    @SuppressLint("DefaultLocale")
    void bind(GameStats gameStats) {
      pts.setText(String.format("%d", gameStats.points));
      reb.setText(String.format("%d", gameStats.rebounds));
      ast.setText(String.format("%d", gameStats.assists));
      blk.setText(String.format("%d", gameStats.blocks));
      stl.setText(String.format("%d", gameStats.steals));
      to.setText(String.format("%d", gameStats.turnovers));

      twoPT.setText(String.format("%d - %d", gameStats.twoPtsMade, gameStats.twoPtsAttempted));
      threePT.setText(
          String.format("%d - %d", gameStats.threePtsMade, gameStats.threePtsAttempted));
      onePT.setText(
          String.format("%d - %d", gameStats.freeThrowMade, gameStats.freeThrowAttempted));
    }
  }
}

File: app/src/main/java/com/dancc/basketballstatskeeper/adapter/StatsAdapter.kt
package com.dancc.basketballstatskeeper.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.dancc.basketballstatskeeper.R.layout
import com.dancc.basketballstatskeeper.adapter.StatsAdapter.StatsViewHolder
import com.dancc.basketballstatskeeper.model.GameStats
import kotlinx.android.synthetic.main.stat_row_header_view.view.ast
import kotlinx.android.synthetic.main.stat_row_header_view.view.blk
import kotlinx.android.synthetic.main.stat_row_header_view.view.onePT
import kotlinx.android.synthetic.main.stat_row_header_view.view.pts
import kotlinx.android.synthetic.main.stat_row_header_view.view.reb
import kotlinx.android.synthetic.main.stat_row_header_view.view.stl
import kotlinx.android.synthetic.main.stat_row_header_view.view.threePT
import kotlinx.android.synthetic.main.stat_row_header_view.view.turnOver
import kotlinx.android.synthetic.main.stat_row_header_view.view.twoPT

class StatsAdapter : Adapter<StatsViewHolder>() {

  private var gameStats = mutableListOf<GameStats>()

  fun setGameStats(gameStats: List<GameStats>) {
    this.gameStats.clear()
    this.gameStats.addAll(gameStats)
    notifyDataSetChanged()
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StatsViewHolder {
    val v = LayoutInflater.from(parent.context)
        .inflate(layout.stat_row_header_view, parent, false)
    return StatsViewHolder(v)
  }

  override fun onBindViewHolder(holder: StatsViewHolder, position: Int) {
    holder.bind(gameStats[position])
  }

  override fun getItemCount(): Int {
    return gameStats.size
  }

  class StatsViewHolder(v: View?) : ViewHolder(v!!) {

    fun bind(gameStats: GameStats) {
      itemView.apply {
        pts.text = String.format("%d", gameStats.points)
        reb.text = String.format("%d", gameStats.rebounds)
        ast.text = String.format("%d", gameStats.assists)
        blk.text = String.format("%d", gameStats.blocks)
        stl.text = String.format("%d", gameStats.steals)
        turnOver.text = String.format("%d", gameStats.turnovers)
        twoPT.text = String.format("%d - %d", gameStats.twoPtsMade, gameStats.twoPtsAttempted)
        threePT.text = String.format("%d - %d", gameStats.threePtsMade, gameStats.threePtsAttempted)
        onePT.text = String.format("%d - %d", gameStats.freeThrowMade, gameStats.freeThrowAttempted)
      }
    }
  }
}